relay-replays/src/recording/mod.rs

```diff
@@ -1,487 +1,134 @@
-        .ok_or(RecordingParseError::Message("no headers found"))?;
-
-    let body = match split.next() {
-        Some(b"") | None => return Err(RecordingParseError::Message("no body found")),
-        Some(body) => body,
-    };
-
-    let mut events = deserialize_compressed(body, limit)?;
-    strip_pii(&mut events).map_err(RecordingParseError::ProcessingAction)?;
-    let out_bytes = serialize_compressed(events)?;
-    Ok([header.into(), vec![b'\n'], out_bytes].concat())
-}
-
-fn deserialize_compressed(
-    zipped_input: &[u8],
-    limit: usize,
-) -> Result<Vec<Event>, RecordingParseError> {
-    let decoder = ZlibDecoder::new(zipped_input);
-
-    let mut buffer = Vec::new();
-    decoder.take(limit as u64).read_to_end(&mut buffer)?;
-
-    Ok(serde_json::from_slice(&buffer)?)
-}
-
-fn serialize_compressed(rrweb: Vec<Event>) -> Result<Vec<u8>, RecordingParseError> {
-    let buffer = serde_json::to_vec(&rrweb)?;
+mod transcoder;
 
-    let mut encoder = ZlibEncoder::new(Vec::new(), Compression::default());
-    encoder.write_all(&buffer)?;
-    let result = encoder.finish()?;
-    Ok(result)
-}
-
-fn strip_pii(events: &mut Vec<Event>) -> Result<(), ProcessingAction> {
-    let mut pii_config = PiiConfig::default();
-    pii_config.applications =
-        BTreeMap::from([(SelectorSpec::And(vec![]), vec!["@common".to_string()])]);
-
-    let pii_processor = PiiProcessor::new(pii_config.compiled());
-    let mut processor = RecordingProcessor::new(pii_processor);
-    processor.mask_pii(events)?;
-
-    Ok(())
-}
-
-// Error
+use transcoder::StringTranscoder;
 
 #[derive(Debug)]
-pub enum RecordingParseError {
+pub enum ParseRecordingError {
     Json(serde_json::Error),
-    Compression(std::io::Error),
     Message(&'static str),
-    ProcessingAction(ProcessingAction),
 }
 
-impl Display for RecordingParseError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+impl fmt::Display for ParseRecordingError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            RecordingParseError::Json(serde_error) => write!(f, "{serde_error}"),
-            RecordingParseError::Compression(io_error) => write!(f, "{io_error}"),
-            RecordingParseError::Message(message) => write!(f, "{message}"),
-            RecordingParseError::ProcessingAction(action) => write!(f, "{action}"),
+            ParseRecordingError::Json(serde_error) => write!(f, "{serde_error}"),
+            ParseRecordingError::Message(message) => write!(f, "{message}"),
         }
     }
 }
 
-impl std::error::Error for RecordingParseError {}
-
-impl From<serde_json::Error> for RecordingParseError {
-    fn from(err: serde_json::Error) -> Self {
-        RecordingParseError::Json(err)
-    }
-}
-
-impl From<std::io::Error> for RecordingParseError {
-    fn from(err: std::io::Error) -> Self {
-        RecordingParseError::Compression(err)
-    }
-}
-
-// Recording Processor
-
-struct RecordingProcessor<'a> {
```

Comment:
The recording processor does not exist anymore. Instead of walking a structure, we now call `scrub_replay`.